tests/providers/amazon/aws/deferrable/hooks/test_batch_client.py

```diff
@@ -0,0 +1,213 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import sys
+
+import botocore
+import pytest
+
+from airflow.exceptions import AirflowException
+from airflow.providers.amazon.aws.hooks.batch_client import BatchClientAsyncHook
+
+if sys.version_info < (3, 8):
+    # For compatibility with Python 3.7
+    from asynctest import mock as async_mock
+else:
+    from unittest import mock as async_mock
+
+pytest.importorskip("aiobotocore")
+
+
+class TestBatchClientAsyncHook:
```

Comment:
Thanks for adding all these tests, they all look good to me. However, for code clarity purposes, could you try to merge some using [@pytest.mark.parametrize](https://docs.pytest.org/en/6.2.x/parametrize.html)? Many of your tests are really similar and I am sure by using `@pytest.mark.parametrize`, you could reduce the length of your unit tests by a lot

Reply:
Even I had parameterised the test at first but got this error https://github.com/Martiusweb/asynctest/issues/152#issue-666326338 and CI was failing, by looking at this PR comment https://github.com/apache/airflow/pull/29801#discussion_r1125620257 I had them separated then the CI was passing for me.

Reply:
Interesting! Thanks for the explanations, I did not know that. LGTM then